doc/HACKING/design/00-overview.md

```diff
@@ -36,36 +36,60 @@ same codebase: the Tor process will run as a client, relay, or authority
 depending on its configuration.
 
 Tor has a few major dependencies, including Libevent (used to tell which
-sockets are readable and writable), OpenSSL (used for many encryption
+sockets are readable and writable), OpenSSL or NSS (used for many encryption
 functions, and to implement the TLS protocol), and zlib (used to
 compress and uncompress directory information).
 
 Most of Tor's work today is done in a single event-driven main thread.
 Tor also spawns one or more worker threads to handle CPU-intensive
-tasks.  (Right now, this only includes circuit encryption.)
+tasks.  (Right now, this only includes circuit encryption and the more
+expensive compression algorithms.)
 
 On startup, Tor initializes its libraries, reads and responds to its
 configuration files, and launches a main event loop.  At first, the only
 events that Tor listens for are a few signals (like TERM and HUP), and
 one or more listener sockets (for different kinds of incoming
-connections).  Tor also configures a timer function to run once per
-second to handle periodic events.  As Tor runs over time, other events
-will open, and new events will be scheduled.
+connections).  Tor also configures several timers to handle periodic
+events.  As Tor runs over time, other events will open, and new events
+will be scheduled.
 
-The codebase is divided into a few main subdirectories:
-
-   src/common -- utility functions, not necessarily tor-specific.
-
-   src/or -- implements the Tor protocols.
-
-   src/test -- unit and regression tests
+The codebase is divided into a few top-level subdirectories, each of
+which contains several sub-modules.
 
    src/ext -- Code maintained elsewhere that we include in the Tor
    source distribution.
 
-   src/trunnel -- automatically generated code (from the Trunnel)
+   src/lib -- Lower-level utility code, not necessarily tor-specific.
+
+   src/trunnel -- Automatically generated code (from the Trunnel)
    tool: used to parse and encode binary formats.
 
+   src/core -- Networking code that is implements the central parts of
+   the Tor protocol and main loop.
+
+   src/feature -- Aspects of Tor (like directory management, running a
+   relay, running a directory authorities, managing a list of nodes,
+   running and using onion services) that are built on top of the
+   mainloop code.
+
+   src/app -- Highest-level functionality; responsible for setting up
+   and configuring the Tor project, making sure all the lower-level
```

Comment:
Hm, Ithink using "The Tor Project" here becomes confusing. What about "The Tor daemon" ?